pkg/controller/controller.go

```diff
@@ -507,7 +508,7 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		// the in-tree provisioner name in case of CSI migration scenarios. The annotation will be
 		// set to the CSI provisioner name by PV controller for migration scenarios
 		// so that external provisioner can correctly pick up the PVC pointing to an in-tree plugin
-		if sc.Provisioner == p.supportsMigrationFromInTreePluginName {
+		if hasMigrated(p, sc) {
```

Comment:
Is there anywhere else that we call this function? If not, let's keep it as it is. I think from a readability perspective, it can be more straightforward